src/models/AssetRewards.ts

```diff
@@ -1,52 +1,58 @@
 export default class AssetRewards {
-  private totalPrice: number
+    private totalPrice: number
 
-  private rewards: Map<string, number>
+    private rewards: Map<string, number>
 
-  public constructor()
+    public constructor()
 
-  public constructor(_rewards: Map<string, number>)
+    public constructor(_rewards: Map<string, number>)
 
-  public constructor(address: string, amount: number)
+    public constructor(address: string, amount: number)
 
-  public constructor(..._params: any[]) {
-    this.totalPrice = 0
-    this.rewards = new Map()
+    public constructor(address: string, amount: number, addressMarketplace: string, amountMarketplace: number)
 
-    if (_params.length === 1) {
-      this.rewards = _params[0]
-      this.rewards.forEach(
-        (v, k) => (this.totalPrice = v + this.totalPrice)
-      )
-    } else if (_params.length === 2) {
-      this.rewards.set(_params[0], _params[1])
-      this.totalPrice = _params[1]
+    public constructor(..._params: any[]) {
+        this.totalPrice = 0
+        this.rewards = new Map()
+
+        if (_params.length === 1) {
```

Comment:
I think giving _params.length === 1 a name, would make this more readable
`const isOneRecipient(or whatever it refers to) = _params.length === 1`

Reply:
Fair point, I'm not even sure what is the case for `_params.length === 1`.
@aaitor or @r-marques can answer this maybe?